chore: allow controlling highlight priority of the cursor line

volskaya opened the pull request:
Right now the cursor line seems to replace all component highlights, due to that high priority, so a field to control it would be really nice.

Will help out people migrating from company, using background colors on kind icons was pretty common.

<img width="700" alt="Screenshot 2025-04-03 at 19 09 22" src="https://github.com/user-attachments/assets/925d59b9-432b-4292-86eb-a70aa7a0f0ac" />

volskaya commented:
Would help https://github.com/Saghen/blink.cmp/issues/1254#issuecomment-2753987613 as well.

Saghen commented:
I've added support for returning a priority from the highlight function in https://github.com/Saghen/blink.cmp/commit/bf2b10e85fe08ba7891ae72802e818fe9af05f9b and set the default kind_icon highlight to 20000 in https://github.com/Saghen/blink.cmp/commit/f1e1940a31f4b748559851a7adca43506a55a991. I've also added a recipe for this type of configuration in https://github.com/Saghen/blink.cmp/commit/1298cc0d670eedca51097abb356b61a43d5b6a26. I think this should allow us to continue hard coding the 10000 priority on cursor line. But, lmk if there's still a use case for this, and I'd be happy to revisit it!

volskaya commented:
> I've added support for returning a priority from the highlight function in [bf2b10e](https://github.com/Saghen/blink.cmp/commit/bf2b10e85fe08ba7891ae72802e818fe9af05f9b) and set the default `kind_icon` highlight to 20000 in [f1e1940](https://github.com/Saghen/blink.cmp/commit/f1e1940a31f4b748559851a7adca43506a55a991). I've also added a recipe for this type of configuration in [1298cc0](https://github.com/Saghen/blink.cmp/commit/1298cc0d670eedca51097abb356b61a43d5b6a26). I think this should allow us to continue hard coding the 10000 priority on cursor line. But, lmk if there's still a use case for this, and I'd be happy to revisit it!

Works great, thank you!
